internal/pruner/pruner.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2015 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package pruner
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/syncthing/protocol"
+	"github.com/syncthing/syncthing/internal/db"
+)
+
+type Pruner struct {
+	dirPatterns  []string
+	filePatterns []string
+}
+
+func New(patterns []string) *Pruner {
+	p := &Pruner{
+		dirPatterns:  make([]string, 0),
```

Comment:
Generally, leaving the slices as nil (here, and in the config) ought to be fine?